### ISSUE TITLE
Links do not work properly with stacks nested inside a tab navigator

### ISSUE THREAD

**Issue by d4rky-pl** [expo/router#797]
### Which package manager are you using? (Yarn is recommended)

yarn

### Summary

Given a layout with Tabs where each Tab is a Stack navigator, when adding `Link` components to the top level `_layout`, the links initially work but then stop working after the tab is navigated to at least once.

Observed when testing on web, `useRouter().push` is also affected.

I've debugged so far that on the first click `isMovingToSiblingRoute` in `global-state/routing.ts:107` is `false` but on the next one it's `true` and the navigation no longer works because it tries to navigate the inner stack navigator and not the outer tabs navigator.

#### Demo recording

https://github.com/expo/router/assets/284395/fbf90b0e-a020-4601-a598-1dcba0ac92d8

### Minimal reproducible example

Here's the repository with a project that exhibits this bug:
https://github.com/d4rky-pl/expo-router-navigation-bug

I tried to create as minimal reproduction as possible

**Comment by d4rky-pl** [comment]
I just mounted the list of links on a page inside one of the stack screens and it still doesn't work so I'm not sure it's related to the parent navigator after all, it seems that links in general do not work well if you have stack navigators nested inside tabs like this.

Is the setup I'm trying to do something wrong or unexpected? If so, what should be the right way to approach this?

**Comment by kennetpostigo** [comment]
Duplicate, and heres some context: https://github.com/expo/router/issues/768#issuecomment-1644925647

**Comment by d4rky-pl** [comment]
@kennetpostigo not convinced it's the same bug. The repository with the repro of the bug does not use `unstable_settings` anywhere and it doesn't throw the error

**Comment by d4rky-pl** [comment]
To unblock myself I finally gave up on making Tabs and Stacks work together properly (which was already painful in 1.x and is completely broken in 2.x) and I switched to just having one `_layout` in the project with `<Slot/>` in it and avoid any sort of navigator nesting completely. I won't have any nice animated transitions between things but after struggling for two months I just want my project to run at all.

**Comment by dzolt** [comment]
I can confirm it also doesn't work on mobile. When i refresh the app i can do router.push() but then the button is not responding. My solution for now was to swap push() with replace() method and it's working every time but it is indeed an issue.

**Comment by janusqa** [comment]
I think I have this issue and neither push nor replace works.
The navigation seems to work the very first time, and then after that it just seems to reload the page being navigated from.

**Comment by minxd95** [comment]
<img width="207" alt="스크린샷 2023-08-14 오전 5 42 33" src="https://github.com/expo/router/assets/61887650/620c5730-1031-4e34-872b-be4693189f39">

Try this architecture. It's not satisfying because I had to place Tab-Navigator inside "tabs" folder. But It works fine.

**Comment by michalstruck** [comment]
I'm facing the same issue, I'll try to recreate the layout with `Slots` as suggested by @d4rky-pl 
It's a shame :/

**Comment by amen6** [comment]
Updating expo-router to 2.0.4 will fix this bug @janusqa @Seimodei

**Comment by janusqa** [comment]
@amen6 how did you manage to install 2.0.4, nothing I do will move me beyond 2.0.0
"npx expo install expo-router@latest" or "npx expo install expo-router@2.0.4" does not install a 2.0.4 it remains at 2.0.0

**Comment by philiplindberg** [comment]
I just tried running the [repro](https://github.com/d4rky-pl/expo-router-navigation-bug) @d4rky-pl provided with expo-router 2.0.4, but I'm still seeing the same issue.

**Comment by janusqa** [comment]
I force installed 2.0.4 with "npm install expo-router"
it gives this when starting the project
```
Some dependencies are incompatible with the installed expo version:
  expo-router@2.0.4 - expected version: 2.0.0
Your project may not work correctly until you install the correct versions of the packages.
```
However it seems to have addressed the issue I was having.

**Comment by amen6** [comment]
@janusqa "npm i expo-router@2.0.4"

**Comment by d4rky-pl** [comment]
@amen6 @janusqa [the latest released version in npm is 2.0.5 now](https://www.npmjs.com/package/expo-router?activeTab=versions). I can confirm that with 2.0.5 the problem presented in my repro is mostly solved. The link to the root path (`/`) is still broken but I believe it's a separate bug (#365)
